elastic/kibana
# [Discover] Fix sorting by _score behavior

## Summary

The integration of EuiDataGrid made it necessary to allow the user to un-sort Discover's document table. (#85561) Default sorting of index patterns without time field sorted internally still by `_score`, but when the user added the `_score` column, there was no indication that's it's sorted. This PR solves it by automatically adding sorting by `_score` when the user selects the column, if she/he didn't sort it before by a different field. This also works when adding _score using an index pattern with timefield and it's in an unsorted state. Also this PR increased test coverage

### Testing

* Create an index pattern without timefield in Discover and make use of it 
* Add `_score` column 
* If no other fields was selected before to sort by, you it should work like this:

https://user-images.githubusercontent.com/463851/108706975-b49bca00-750f-11eb-8391-c93fbdc06757.mp4



### Checklist
